clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -326,6 +340,7 @@ private void pollSelectionKeys(Iterable<SelectionKey> selectionKeys,
             SelectionKey key = iterator.next();
             iterator.remove();
             KafkaChannel channel = channel(key);
+            long channelStartTimeNanos = recordTimePerConnection ? time.nanoseconds() : 0;
```

Comment:
@junrao @ijuma I ran `ProducerPerformance` and `ConsumerPerformance` on my laptop and didn't see any noticeable difference. This is the throughput in MB/s (average of three runs):

Test     (message size)    | trunk             |  with PR
-------------------------|--------------|------------
Producer (100 bytes)      | 158.68          | 160.18 
Producer (1000 bytes)    | 355.43         | 350.36
Consumer (100 bytes)    | 376.20          | 378.52 
Consumer (1000 bytes)  | 559.45          | 559.45